distribute.go

```diff
@@ -232,18 +232,32 @@ type distRunner struct {
 	d          *distributer
 }
 
-func (r *distRunner) Run(ctx context.Context, inp, out chan Dataset) error {
+func (d *distRunner) Equals(other interface{}) bool {
+	r, ok := other.(*distRunner)
+	isEqual := ok && r.Runner.Equals(d.Runner) && d.MasterAddr == r.MasterAddr
+	if !isEqual || len(d.Addrs) != len(r.Addrs) {
+		return false
+	}
+	for i, v := range d.Addrs {
+		if v != r.Addrs[i] {
+			return false
+		}
+	}
+	return true
```

Comment:
no need to check Addrs :)
also, string before runner
```suggestion
	o, ok := other.(*distRunner)
	return ok &&  r.MasterAddr == o.MasterAddr && r.Runner.Equals(o.Runner)
```